<a href="https://colab.research.google.com/github/mukedon/python_studies/blob/main/tutorials/3_f_strings.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Formatted String (f-string)

In [20]:
"""
In case we want to change a string,
we put it into a formatted string
"""
f_temp = 21
f'The temperature is {f_temp} degrees Celcius'

'The temperature is 21 degrees Celcius'

With this we can make changes on it.

In [21]:
f'The temperature is {(f_temp * 1.8) + 32} degrees Fahrenheit'

'The temperature is 69.80000000000001 degrees Fahrenheit'

### Rounding with f-strings


In [22]:
f'The temperature is {(f_temp * 1.8) + 32:.2f} degrees Fahrenheit'

'The temperature is 69.80 degrees Fahrenheit'

In [23]:
f'The temperature is {(f_temp * 1.8) + 32:.2e} degrees Fahrenheit'

'The temperature is 6.98e+01 degrees Fahrenheit'

In [24]:
f'The temperature is {(f_temp * 1.8) + 32:.2%} degrees Fahrenheit'

'The temperature is 6980.00% degrees Fahrenheit'

### Big numbers with f-strings

In [15]:
f_temp_sun = 27000000

In [25]:
f'The temperature is {(f_temp_sun * 1.8) + 32:.2f} degrees Fahrenheit'

'The temperature is 48600032.00 degrees Fahrenheit'

In [26]:
f'The temperature is {(f_temp_sun * 1.8) + 32:,.2f} degrees Fahrenheit'

'The temperature is 48,600,032.00 degrees Fahrenheit'

### More formatting with f-strings

In [28]:
f_temp_earth = 21

print(f'The temperature is {(f_temp_sun * 1.8) + 32:,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:,.2f} degrees Fahrenheit')

The temperature is 48,600,032.00 degrees Fahrenheit
The temperature is 69.80 degrees Fahrenheit


:[width],.[decimals][type]

In [29]:
print(f'The temperature is {(f_temp_sun * 1.8) + 32:15,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:15,.2f} degrees Fahrenheit')

The temperature is   48,600,032.00 degrees Fahrenheit
The temperature is           69.80 degrees Fahrenheit


We set [width] as selecting $15$, which helps the readability, now each value: $48,600,032.00$ and $69.80$ has the same width (space): $15$.

:[alignment][width],.[decimals][type]

In [30]:
print(f'The temperature is {(f_temp_sun * 1.8) + 32:<15,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:<15,.2f} degrees Fahrenheit')

The temperature is 48,600,032.00   degrees Fahrenheit
The temperature is 69.80           degrees Fahrenheit


Aligned them to left.

In [31]:
print(f'The temperature is {(f_temp_sun * 1.8) + 32:^15,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:^15,.2f} degrees Fahrenheit')

The temperature is  48,600,032.00  degrees Fahrenheit
The temperature is      69.80      degrees Fahrenheit


Aligned them to centre.

In [32]:
print(f'The temperature is {(f_temp_sun * 1.8) + 32:>15,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:>15,.2f} degrees Fahrenheit')

The temperature is   48,600,032.00 degrees Fahrenheit
The temperature is           69.80 degrees Fahrenheit


Aligned them to right again.

:[fill][alignment][width],.[decimals][type]

In [33]:
print(f'The temperature is {(f_temp_sun * 1.8) + 32:*^15,.2f} degrees Fahrenheit')
print(f'The temperature is {(f_temp_earth * 1.8) + 32:*^15,.2f} degrees Fahrenheit')

The temperature is *48,600,032.00* degrees Fahrenheit
The temperature is *****69.80***** degrees Fahrenheit


Filled empty spaces with `*`.